mongo/options/serverapioptions.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package options
+
+import "fmt"
+
+// ServerAPIOptions represents options used to configure the API version sent to the server
+// when running commands.
+//
+// Sending a specified server API version causes the server to behave in a manner compatible with that
+// API version. It also causes the driver to behave in a manner compatible with the driver’s behavior as
+// of the release when the driver first started to support the specified server API version.
+//
+// The user must specify a ServerAPIVersion if including ServerAPIOptions in their client. That version
+// must also be currently supported by the driver.
+type ServerAPIOptions struct {
```

Comment:
To match the other options types, the fields of this struct should be pointers. There should also be setters for each field so we can have builder-style construction. See the `AutoEncryptionOptions` type for an example.

Reply:
As discussed offline, I changed the two bools to be pointers and left `ServerAPIVersion` as is because it's just a string. Added the setters in (had not included them because the spec wanted the `ServerAPIOptions` object to be "immutable", but looking at setters on other types it seems fine).